### PR TITLE
[DF] Fix ROOT-9632: do not throw if define return type is unknown to the interpreter

### PR DESCRIPTION
**Pull request by dpiparo**
but add a comment to the jitted code in order to obtain a clear error message if
the user tries to use this type in a jitted action/transformation later in the chain.

### CODE REVIEW

**Comment by bluehood** [comment]
So before:
* cling does not know the type, type is not used in jitting: throw
* cling does not know the type, type is used in jitting: throw

while now:
* cling does not know the type, type is not used in jitting: nothing happens at all
* cling does not know the type, type is used in jitting: ?

File: tree/dataframe/inc/ROOT/RDFInterface.hxx [hunk 1]
```diff
@@ -1744,11 +1744,17 @@ private:
...
-         throw std::runtime_error(msg);
+         //const auto msg =
+         //   "Return type of Define expression was not understood. Type was " + std::string(typeid(RetType).name());
+         //throw std::runtime_error(msg);
+         const std::string tName(typeid(RetType).name());
+         retTypeName = "unknown_type_" +
+                       tName +
+                       "/* --> Did you forget to inject \"" +
+                       tName +
+                       "\"'s definition in the interpreter?*/";
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDFInterface.hxx]: how does this not break the `gInterpreter->Declare(retTypeDeclaration.c_str());` below?

Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDFInterface.hxx]: so the `Declare` below will try to jit something like
```
"namespace __tdf1 { using x_type = uknown_type_typeidnameoftype; };"
```

How does that `Declare` succeed if `unknown_type_typeidnameoftype` is not a type?

**Comment by dpiparo** [comment]
Example effect of the new, complete implementation:
```.cpp
#include "ROOT/RDataFrame.hxx"

struct DecayAngles {
  Double_t phi{0.0};
};


int main()
{
  ROOT::RDataFrame d( 10 );
  auto dnotype = d.Define( "foo", [](){ return DecayAngles(); } );
  *dnotype.Count(); // does work
//  *dnotype.Filter("foo.phi > 0").Count(); // does not work and prints nice error message
  return 0;
}
```
error message
```
input_line_15:1:56: error: variable has incomplete type '__tdf1::foo_type' (aka '__tdf1::DecayAngles')
namespace __tdf_0{ auto tdf_f = []() {__tdf1::foo_type foo;
                                                       ^
input_line_13:1:26: note: forward declaration of '__tdf1::DecayAngles'
namespace __tdf1 { class DecayAngles;/* Did you forget to declare type DecayAngles in the interpreter?*/ using foo_type = DecayAngles; }
                         ^
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Cannot interpret the following expression:
foo.phi > 0

Make sure it is valid C++.
Abort trap: 6
```

**Pull request merged**